### PR TITLE
[azquery] add metrics batch feature

### LINKED ISSUE #12252: Support Querying metrics for multiple resources
### Support Querying metrics for multiple resources

This may well already be possible, however, I was not able to find or understand.

Is it possible to get metrics of multiple resources in a single call/query?

For eg. metrics for a list of VMs.
or 
metrics for set of VMs of a given resource-group / subscription?

### LINKED ISSUE #20314: [Monitor Query] Migrate to data plane metrics query API
As part of the migration away from the ARM API, we'll need to add API surface to support batch metrics queries. Up to 50 resource IDs in the same subscription and region can be queried in a single API call.

### PR DESCRIPTION
**Pull request by gracewilcox**
Closes #20314
Fixes #12252

Add `MetricsBatchClient` to azquery which allows users to send batch requests for metric data.

### CODE REVIEW

File: sdk/monitor/azquery/metricsbatch_client._test.go
Comment (jhendrixMSFT) [inline sdk/monitor/azquery/metricsbatch_client._test.go:1]: Extra `.` in file name.

File: sdk/monitor/azquery/models.go [hunk 1]
```diff
@@ -224,28 +224,65 @@ type MetricNamespaceName struct {
 	MetricNamespaceName *string
 }
 
+// MetricResultsResponse - The metrics result for a resource.
+type MetricResultsResponse struct {
+	// The collection of metric data responses per resource, per metric.
+	Values []*MetricResultsResponseValuesItem
+}
+
+type MetricResultsResponseValuesItem struct {
```
Comment (jhendrixMSFT) [inline sdk/monitor/azquery/models.go:233]: I think we should shorten this type name (I mentioned this on the APIView). I suggested `MetricValues` or `MetricData` but open to suggestions.

Comment (gracewilcox) [reply sdk/monitor/azquery/models.go:233]: I agree, it's a little too long. Changing it to MetricValues

Comment (gracewilcox) [reply sdk/monitor/azquery/models.go]: fixed in [d4ede52](https://github.com/Azure/azure-sdk-for-go/pull/21996/commits/d4ede52883798e7427f4a63972949b0568145ff5)

File: sdk/monitor/azquery/metricsbatch_client._test.go
Comment (gracewilcox) [reply sdk/monitor/azquery/metricsbatch_client._test.go:1]: fixed in [d4ede52](https://github.com/Azure/azure-sdk-for-go/pull/21996/commits/d4ede52883798e7427f4a63972949b0568145ff5)

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**